renku/api/models/run.py

```diff
@@ -58,17 +59,30 @@
 )
 from renku.core.plugins.provider import RENKU_ENV_PREFIX
 
+name_pattern = re.compile("[a-zA-Z0-9-_]+")
+
 
 class _PathBase(PathLike):
     @ensure_project_context
-    def __init__(self, path, project):
-        self._path = Path(path)
+    def __init__(self, name: str, path: Union[str, Path], project):
+        if name and not name_pattern.match(name):
+            raise errors.ParameterError(
+                f"Name {name} contains illegal characters. Only characters, numbers, _ and - are allowed."
+            )
+        self.name = name
+
+        env_value = environ.get(f"{RENKU_ENV_PREFIX}{name}", None)
```

Comment:
What happens if we have multiple runs, executing in parallel (e.g. a `workflow iterate`)? I'd expect that the execution environments to be isolated, but do we know that for sure for all of the providers?